sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -1966,6 +1966,51 @@ pi_result cuda_piDeviceGetInfo(pi_device device, pi_device_info param_name,
                         name.data());
   }
 
+  case PI_DEVICE_INFO_MAX_MEM_BANDWIDTH: {
+    int major = 0;
+    sycl::detail::pi::assertion(
+        cuDeviceGetAttribute(&major,
+                             CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR,
+                             device->get()) == CUDA_SUCCESS);
+
+    int minor = 0;
+    sycl::detail::pi::assertion(
+        cuDeviceGetAttribute(&minor,
+                             CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR,
+                             device->get()) == CUDA_SUCCESS);
+
+    // Some specific devices seem to need special handling. See reference
+    // https://github.com/jeffhammond/HPCInfo/blob/master/cuda/gpu-detect.cu
+    bool is_xavier_agx = major == 7 && minor == 2;
+    bool is_orin_agx = major == 8 && minor == 7;
+
+    int memory_clock_khz = 0;
+    if (is_xavier_agx) {
+        memory_clock_khz = 2133000;
+    } else if (is_orin_agx) {
+        memory_clock_khz = 3200000;
+    } else {
+        sycl::detail::pi::assertion(
+            cuDeviceGetAttribute(&memory_clock_khz, CU_DEVICE_ATTRIBUTE_MEMORY_CLOCK_RATE,
+                                device->get()) == CUDA_SUCCESS);
+    }
+    
+    int memory_bus_width = 0;
+    if (is_orin_agx) {
+       memory_bus_width = 256;
+    } else {
+        sycl::detail::pi::assertion(
+            cuDeviceGetAttribute(&memory_bus_width,
+                                CU_DEVICE_ATTRIBUTE_GLOBAL_MEMORY_BUS_WIDTH,
+                                device->get()) == CUDA_SUCCESS);
+    }
+
+    uint64_t memory_bandwidth = uint64_t(memory_clock_khz) * memory_bus_width * 250;
```

Comment:
Sorry this is post-approval, but what does the 250 represent here?

Reply:
250 = 2 * 125 

2 is dual data rate because Jeff finds that HBM is a DDR memory. The value of "memory_bus_width" is memory bus width in bits.